basic/src/test/java/org/jboss/hal/testsuite/test/configuration/undertow/AJPListenerTestCase.java

```diff
@@ -64,136 +62,143 @@
+    private static final String SOCKET_BINDING_AJP_TBR = "socket_binding-ajp-tbr_" + RandomStringUtils.randomAlphanumeric(5);
+    private static final String SOCKET_BINDING_AJP_TBA = "socket_binding-ajp-tbr_" + RandomStringUtils.randomAlphanumeric(5);
 
     @BeforeClass
-    public static void setUp() throws IOException, CommandFailedException, TimeoutException, InterruptedException {
-        httpServer = operations.createHTTPServer();
-        ajpListener = operations.createAJPListener(httpServer);
-        ajpListenerToBeRemoved = operations.createAJPListener(httpServer);
-        address = ajpListenerTemplate.resolve(context, httpServer, ajpListener);
+    public static void setUp() throws Exception {
+        undertowOps.createSocketBinding(SOCKET_BINDING_AJP);
+        undertowOps.createSocketBinding(SOCKET_BINDING_AJP_TBR);
+        undertowOps.createSocketBinding(SOCKET_BINDING_AJP_TBA);
+        undertowOps.createSocketBinding(REDIRECT_SOCKET_VALUE);
+        //add http server
+        operations.add(HTTP_SERVER_ADDRESS);
+        administration.reloadIfRequired();
+        //add ajp listeners which will be used
+        client.apply(new AddUndertowListener.AjpBuilder(AJP_LISTENER, HTTP_SERVER, SOCKET_BINDING_AJP)
+                .enabled(true)
+                .build());
+        client.apply(new AddUndertowListener.AjpBuilder(AJP_LISTENER_TBR, HTTP_SERVER, SOCKET_BINDING_AJP_TBR)
+                .enabled(true)
+                .build());
+
+        administration.reloadIfRequired();
     }
 
     @Before
     public void before() {
-        navigate2ajpListener();
+        page.navigate();
+        page.viewHTTPServer(HTTP_SERVER)
+                .switchToAJPListeners()
+                .selectItemInTableByText(AJP_LISTENER);
     }
 
     @AfterClass
-    public static void tearDown() throws InterruptedException, CommandFailedException, TimeoutException, IOException {
-        operations.removeAJPListener(httpServer, ajpListener);
-        operations.removeHTTPServer(httpServer);
+    public static void tearDown() throws InterruptedException, CommandFailedException, TimeoutException, IOException, OperationException {
+        operations.removeIfExists(AJP_LISTENER_ADDRESS);
+        operations.removeIfExists(AJP_LISTENER_ADDRESS_TBA);
+        operations.removeIfExists(AJP_LISTENER_ADDRESS_TBR);
+        //remove used http server
+        operations.removeIfExists(HTTP_SERVER_ADDRESS);
+        //remove used socket bindings
+        undertowOps.removeSocketBinding(REDIRECT_SOCKET_VALUE);
+        undertowOps.removeSocketBinding(SOCKET_BINDING_AJP);
+        undertowOps.removeSocketBinding(SOCKET_BINDING_AJP_TBA);
+        undertowOps.removeSocketBinding(SOCKET_BINDING_AJP_TBR);
+        //servers should not be in reload or restart required after test case ends
+        administration.reloadIfRequired();
+        administration.restartIfRequired();
```

Comment:
If you switch order of `reloadIfRequired()` and `restartIfRequired()` you may save some seconds since reload may not be needed.